### ISSUE TITLE
Arrow functions not aligning properly when nested

### ISSUE THREAD

**Issue by bram-pkg** [FriendsOfPHP/PHP-CS-Fixer#6227]
#### Configuration file

```php
<?php

declare(strict_types=1);

$finder = PhpCsFixer\Finder::create()
    ->ignoreVCSIgnored(true)
    ->ignoreDotFiles(false)
    ->exclude([
        'documentation/',
        'node_modules/',
        'vendor/',
    ])
    ->in(__DIR__);

$config = new PhpCsFixer\Config();
return $config
    ->setRules([
        '@PSR12'                                 => true,
        '@PhpCsFixer'                            => true,
        'binary_operator_spaces'                 => [
            'operators' => [
                '=>' => 'align',
            ],
        ],
        'multiline_whitespace_before_semicolons' => [
            'strategy' => 'no_multi_line',
        ],
    ])
    ->setFinder($finder);

```

### Code snippet that reproduces the problem

```php
$students = $execution->students()
    ->whereHas('tasks', fn (Builder $query): Builder => $query
    ->where('client_id', $examiner->id)
    ->orWhere('supervisor_id', $examiner->id))
    ->get();
```

I think should be:

```php
$students = $execution->students()
    ->whereHas('tasks', fn (Builder $query): Builder => $query
        ->where('client_id', $examiner->id)
        ->orWhere('supervisor_id', $examiner->id))
    ->get();
```

**Comment by bram-pkg** [comment]
For now I fixed this by adding parenthesis around the nested `$query` function.

```php
$students = $execution->students()
    ->whereHas('tasks', fn (Builder $query): Builder => (
        $query
            ->where('client_id', $examiner->id)
            ->orWhere('supervisor_id', $examiner->id))
    ))
    ->get();
```

**Comment by christhofer** [comment]
It's been an issue for quite a while. There's a pull request to fix this but doesn't get merged either.
#5021 
#5587

**Comment by julienfalque** [comment]
Fixed by #5587.
